### PR TITLE
Merge 'dev-awinkler-1' --> 'master'

### PR DESCRIPTION
**Pull request by andrewwinkler**

### CODE REVIEW

File: writingIdomatic.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# /pythonPractice/writingIdomatic.py
+# -*- coding: UTF-8 -*-
+
+from __future__ import print_function
+
+print("\n--Let's Practice!--\n")
+
+# Notes:    When using 'if' statements do the error checking FIRST, then have the code that supposed to run normally,
```
Comment (thomashw) [inline writingIdomatic.py:8]: Definitely!

File: writingIdomatic.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Notes:    Chain arguments in the 'if' statements if you can
+# Ex: None
+
+
+# Notes:    Don’t repeat a '==' comparison in an 'if' statement, use a list or set when you can. ‘x’ in ‘some_list’ can return true or false
+# Ex:
+
+is_generic_name = False
+name = 'Thomas'
+if name == 'Thomas' or name == 'Amir' or name == 'Brett':
```
Comment (thomashw) [inline writingIdomatic.py]: Using constants is good practice if you're typing something more than once. For example, I see 'Thomas' 4 times. If you needed to change that value for whatever reason, using a constant you'd only need to change it in one place.

File: writingIdomatic.py (hunk 1 above)
Comment (andrewwinkler) [reply writingIdomatic.py:8]: Woo

File: writingIdomatic.py (hunk 2 above)
Comment (andrewwinkler) [reply writingIdomatic.py]: Makes sense, updated

File: writingIdomatic.py [hunk 3]
```diff
@@ -17,14 +17,15 @@
 # Notes:    Don’t repeat a '==' comparison in an 'if' statement, use a list or set when you can. ‘x’ in ‘some_list’ can return true or false
 # Ex:
 
+NAME = 'Thomas'
+
 is_generic_name = False
-name = 'Thomas'
-if name == 'Thomas' or name == 'Amir' or name == 'Brett':
+if NAME == 'Thomas' or NAME == 'Amir' or NAME == 'Brett':
```
Comment (thomashw) [inline writingIdomatic.py]: It should be:

```
THOMAS = 'Thomas'
AMIR = 'Amir'
BRETT = 'Brett'
KIERAN = 'Kieran'
ANDREW = 'Andrew'

if name == THOMAS or name == AMIR or name == BRETT
```

You could/should also put an underscore in front of all of the constants to indicate they're internal to the class/file.

File: writingIdomatic.py [hunk 4]
```diff
@@ -17,14 +17,15 @@
...
     is_generic_name = True
 print(is_generic_name) # prints 'True'
 
-name = 'Thomas'
-is_generic_name = name in ('Thomas', 'Amir', 'Brett')
+is_generic_name = False
+is_generic_name = NAME in ('Thomas', 'Amir', 'Brett')
```
Comment (thomashw) [inline writingIdomatic.py]: ```is_generic_name = name in (THOMAS, AMIR, BRETT)```

File: writingIdomatic.py [hunk 5]
```diff
@@ -60,7 +61,7 @@
 #           It is not executed if the loop is ended prematurely
 # Ex:
 
-name_list = ['Andrew', 'Thomas', 'Amir', 'Brett', 'Kieran']
+name_list = ['Andrew', NAME, 'Amir', 'Brett', 'Kieran']
```
Comment (thomashw) [inline writingIdomatic.py]: ```name_list = [ANDREW, THOMAS, AMIR, BRETT, KIERAN]```

File: writingIdomatic.py (hunk 3 above)
Comment (andrewwinkler) [reply writingIdomatic.py]: Makes sense for the constant names
For the underscore, work has told me to use a leading undersore for variables that aren't used. 
An example is the os.popen which returns console height and width but you might only use one.

File: writingIdomatic.py (hunk 4 above)
Comment (andrewwinkler) [reply writingIdomatic.py]: Updated

File: writingIdomatic.py (hunk 5 above)
Comment (andrewwinkler) [reply writingIdomatic.py]: Updated

File: writingIdomatic.py (hunk 3 above)
Comment (thomashw) [reply writingIdomatic.py]: That... is a weird use. They don't use underscores for private variables?

**Comment by andrewwinkler** [comment]
We just use it for ‘private’ methods that are only called by its class or something like that.  
Never seen it used for variables unless it’s the aforementioned reason.

**Comment by thomashw** [comment]
Take a look at libDHC.py (in your code review). There's a couple private variables in there. It's just good practice to tell other developers they aren't intended to be used outside of the class/file.

If you use double underscores Python will do some name mangling but it'll still be accessible outside.

**Comment by andrewwinkler** [comment]
Hmm okay I guess I misunderstood. That makes sense though.  
Thank you!

**Comment by thomashw** [comment]
A lot of languages actually differentiate between public and private variables (like Java) so if you declare it private it actually can't be accessed. With Python it's just style. I like it because otherwise if you're writing code to be consumed by outside developers it's tough for them to know what they should be accessing and what not to.

**Comment by andrewwinkler** [comment]
Hmm yeah makes sense. 
I definitely remember that from Java. 
You’re right though, from at least a readability point of view in python it’s important.

File: writingIdomatic.py (hunk 3 above)
Comment (andrewwinkler) [reply writingIdomatic.py]: https://www.python.org/dev/peps/pep-0008/#constants
We use PEP8 and from what I can tell that formatting style for constants is just all caps with underscores (this is for me)

I won't update with a leading underscores in this script.

**Comment by thomashw** [comment]
I can't find this comment. :| Did you delete it?

File: writingIdomatic.py (hunk 3 above)
Comment (thomashw) [reply writingIdomatic.py]: Ok that's fine. Also, "Use one leading underscore only for non-public methods and instance variables".

That should clear up the confusion about using underscores for instance variables.

Comment (andrewwinkler) [reply writingIdomatic.py]: Ah yes that makes sense. Thanks for pointing that out.

**Pull request merged**